apollo-router-core/src/traits.rs

```diff
@@ -35,7 +35,7 @@ pub trait ServiceRegistry: Send + Sync + Debug {
 pub trait Fetcher: Send + Sync + Debug {
     /// Constructs a stream of responses.
     #[must_use = "streams do nothing unless polled"]
-    async fn stream(&self, request: Request) -> Result<Response, FetchError>;
+    async fn stream(&self, request: &SubgraphRequest) -> Result<RouterResponse, FetchError>;
```

Comment:
Not sure if this should still be called "stream"... :thinking:

Reply:
Yeah i m not comfortable with this name either!

Reply:
I'm still looking at removing Fetcher entirely don't worry